src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCISharedBuildJobQueueService.java

```diff
@@ -406,57 +419,71 @@ private ProgrammingExerciseParticipation retrieveParticipationWithRetry(Long par
+                sendQueuedJobsOverWebsocket(event.getItem().courseId());
+            }
         }
     }
 
     private class ProcessingBuildJobItemListener implements EntryAddedListener<Long, LocalCIBuildJobQueueItem>, EntryRemovedListener<Long, LocalCIBuildJobQueueItem> {
 
         @Override
         public void entryAdded(com.hazelcast.core.EntryEvent<Long, LocalCIBuildJobQueueItem> event) {
-            log.debug("CIBuildJobQueueItem added to processing jobs: {}", event.getValue());
-            localCIBuildQueueWebsocketService.sendRunningBuildJobs(getProcessingJobs());
-            long courseID = event.getValue().getCourseId();
-            localCIBuildQueueWebsocketService.sendRunningBuildJobsForCourse(courseID, getProcessingJobsForCourse(courseID));
+            if (schedulingActive) {
+                log.debug("CIBuildJobQueueItem added to processing jobs: {}", event.getValue());
+                sendProcessingJobsOverWebsocket(event.getValue().courseId());
+            }
         }
 
         @Override
         public void entryRemoved(com.hazelcast.core.EntryEvent<Long, LocalCIBuildJobQueueItem> event) {
-            log.debug("CIBuildJobQueueItem removed from processing jobs: {}", event.getOldValue());
-            localCIBuildQueueWebsocketService.sendRunningBuildJobs(getProcessingJobs());
-            long courseID = event.getOldValue().getCourseId();
-            localCIBuildQueueWebsocketService.sendRunningBuildJobsForCourse(courseID, getProcessingJobsForCourse(courseID));
+            if (schedulingActive) {
+                log.debug("CIBuildJobQueueItem removed from processing jobs: {}", event.getOldValue());
+                sendProcessingJobsOverWebsocket(event.getOldValue().courseId());
+            }
         }
     }
 
-    private class BuildAgentListener implements EntryAddedListener<String, LocalCIBuildAgentInformation>, EntryRemovedListener<String, LocalCIBuildAgentInformation> {
+    public class BuildAgentListener implements EntryAddedListener<String, LocalCIBuildAgentInformation>, EntryRemovedListener<String, LocalCIBuildAgentInformation> {
 
         @Override
         public void entryAdded(com.hazelcast.core.EntryEvent<String, LocalCIBuildAgentInformation> event) {
-            log.debug("Build agent added: {}", event.getValue());
-            localCIBuildQueueWebsocketService.sendBuildAgentInformation(getBuildAgentInformation());
+            if (schedulingActive) {
+                log.debug("Build agent added: {}", event.getValue());
+                localCIBuildQueueWebsocketService.sendBuildAgentInformation(getBuildAgentInformation());
+            }
         }
 
         @Override
         public void entryRemoved(com.hazelcast.core.EntryEvent<String, LocalCIBuildAgentInformation> event) {
-            log.debug("Build agent removed: {}", event.getOldValue());
-            localCIBuildQueueWebsocketService.sendBuildAgentInformation(getBuildAgentInformation());
+            if (schedulingActive) {
```

Comment:
Perhaps create an extra method sendBuildAgentInformationOverWebsocket() as you did in the other Listeners for consistency